corehq/apps/registration/utils.py

```diff
@@ -170,7 +170,7 @@ def request_new_domain(request, project_name, is_new_user=True, is_new_sso_user=
                 'first_domain_for_user': is_new_user,
                 'error': str(error),
             })
-            new_domain.delete()
+            new_domain.delete(leave_tombstone=True)
```

Comment:
Will this make it impossible to create a domain with the same name in the future?

Reply:
@millerdev it's currently impossible, as the domain is never actually deleted due to the `ValueError` being thrown. I'm just trying to fix the error here, but I think taking a different approach (full delete and working around the domain deletion checks) can be done later. I don't have time at the moment to dedicate to this.